pyvera/__init__.py

```diff
@@ -731,12 +716,22 @@ def is_locked(self, refresh=False):
 class VeraThermostat(VeraDevice):
     """Class to represent a thermostat."""
 
-    def get_payload_parameter_name(self, name):
-        return "New" + name
-
     def set_temperature(self, temp):
         """Set current goal temperature / setpoint"""
-        self.set_value('CurrentSetpoint', temp)
+
+        # Set both the heat and cool setpointa
+        self.set_service_value(
+            self.thermostat_cool_setpoint,
+            'CurrentSetpoint',
+            'NewCurrentSetpoint',
+            mode)
+
+        self.set_service_value(
+            self.thermostat_heat_setpoint,
+            'CurrentSetpoint',
+            'NewCurrentSetpoint',
+            mode)
```

Comment:
Thanks very much for taking a look - and fixing the bugs. Hopefully now fixed.

If the bugs are fixed than I'm pretty sure this will do what the old code did (set everything with a `setpoint` to the new value). 

I'm sure this isn't a good solution for a heat/cool heating system - but it probably `sort of` works.

This at least uncovers the issue (which was hidden by the old code). As long as it works as well as it did before - I'd prefer to release.

If you have the time inclination - then it would be great to have another PR to make this correct. This will doubtless need changes to how we interact with HA too.

Does that make sense to you? 

Also be very grateful if you could check the code now works!